internal/services/containers/kubernetes_cluster_network_resource_test.go

```diff
@@ -813,6 +813,38 @@ func TestAccKubernetesCluster_httpProxyConfig(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesCluster_httpProxyConfigUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster", "test")
+	r := KubernetesClusterResource{}
+	noProxy := "\"localhost\", \"127.0.0.1\", \"mcr.microsoft.com\""
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.httpProxyConfig(data, noProxy),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+			ExpectNonEmptyPlan: true,
```

Comment:
Could you explain why we need this here?

Reply:
Hi @stephybun, sorry for late response.

There're tags added to the `azurerm_linux_virtual_machine` by some policies:
```
  ~ tags                                                   = {
                  - "azsecpack"                                                                = "nonprod" -> null
                  - "platformsettings.host_environment.service.platform_optedin_for_rootcerts" = "true" -> null
                }
```

I refers the following codes where it adds "ExpectNonEmptyPlan: true"

https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_network_resource_test.go#L799C2-L799C2